color.go

```diff
@@ -22,12 +22,19 @@ const (
 	White
 )
 
+// ANSIColorPath can be used to override the default ANSIColor style for the given path
+var ANSIColorPath = func(path string) string {
+	return ""
+}
+
 // ANSIColor
 func ANSIColor(node *Node, s string) string {
```

Comment:
The minimum change here would be to turn `ANSIColor` into a function pointer something like

```go
var ANSIColor = func(node *Node, s string) string {
```

Then you can override `ANSIColor` as you please and call its previous value if you don't want to handle it.

Though changes like this should really maybe go in the `Options` struct rather than making them global.

What do you think @a8m

Reply:
> Though changes like this should really maybe go in the Options struct rather than making them global.

Yeah, I find it cleaner to add a `ColorFunc` option to the `Options` struct. We can check if it exists ([here](https://github.com/a8m/tree/blob/14a2569951d04743a926d1bd9d4d5ae04eaa32c1/node.go#L304-L306)) and call it, or use the default one.

Users can also use the default as mentioned by @ncw:
```go
opt.ColorFunc = func(node *Node, s string) string {
	if ... {
		return /* custom */
	}
	return tree.ANSIColor(node, s)
}
```

Reply:
Thanks @ncw @a8m . Sounds good to me, branch updated. One catch, I need the `Node.path` to determine the object type.  Added a func to access that.